gms/gossiper.cc

```diff
@@ -820,6 +820,13 @@ future<semaphore_units<>> gossiper::lock_endpoint_update_semaphore() {
     return get_units(_endpoint_update_semaphore, 1);
 }
 
+future<> gossiper::mutate_live_and_unreachable_endpoints(std::function<void(gossiper&)> func) {
+    auto lock = co_await lock_endpoint_update_semaphore();
+    func(*this);
+    _live_endpoints_version += (_live_endpoints != _shadow_live_endpoints);
```

Comment:
We could increment it unconditionally too, it wouldn't hurt.

Reply:
Initially I did bump it unconditionally, but since it would cause an unnecessary "hiccup" of the failure detector if only _unreachable_endpoints in modified (in `evict_from_membership` and possibly in `real_mark_alive`), I decided to make it more accurate (BTW bumping it in `replicate_live_endpoints_on_change` in the previous version did just that - bump it only on change from shadow_live_endpoints).